components/Modal.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import Link from 'next/link';
+
+const Modal = ({ show, onClose, children }) => {
+  const [isBrowser, setIsBrowser] = useState(false);
+
+  useEffect(() => {
+    setIsBrowser(true);
+  }, []);
+
+  const handleCloseClick = event => {
+    event.preventDefault();
+    onClose();
+  };
+
+  const modalContent = show ? (
+    <StyledModalOverlay>
+      <StyledModal>
+        <Link passHref href="#">
+          <StyledCloseButton onClick={handleCloseClick}>
+            <svg
+              width="14"
+              height="14"
+              viewBox="0 0 14 14"
+              fill="none"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              <path
+                d="M12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41L12.59 0Z"
+                fill="var(--text-lightcolor)"
+              />
+            </svg>
+          </StyledCloseButton>
+        </Link>
+        {children}
+      </StyledModal>
+    </StyledModalOverlay>
+  ) : null;
+
+  if (isBrowser) {
+    return ReactDOM.createPortal(
+      modalContent,
+      document.getElementById('modal-root')
+    );
+  } else {
+    return null;
+  }
+};
+
+const StyledModal = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  position: relative;
+  margin: 3rem 1rem;
+  background: white;
+  width: 90%;
+  max-width: 800px;
+  padding: 5rem 1rem;
+`;
+
+const StyledCloseButton = styled.button`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: absolute;
+  top: 1rem;
+  right: 1rem;
+  border: 1px solid var(--text-lightcolor);
+  height: 1.6rem;
+  width: 1.6rem;
+  color: var(--text-lightcolor);
+`;
+
+const StyledModalOverlay = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
```

Comment:
What do you need this for?

Reply:
the dark background needs that to cover the whole screen